### ISSUE TITLE
Kernel fails at launch : warn() TypeError

### ISSUE THREAD

**Issue by dlesnoff** [sagemath/pari-jupyter#22]
0S: Archlinux
Linux Kernel: 5.16.1
jupyter versions:
IPython: 7.31.0
jupyterlab: 3.2.8
notebook: 6.4.4
pari-jupyter : 1.4.0
python: 3.10

Everytime I open a notebook with pari/gp kernel, I get a crash at launch of the pari/gp kernel.
It tries to restart 5 times before giving up.
I provide my console log (with python syntax highlighting).
```python
[I 22:03:22.994 NotebookApp] KernelRestarter: restarting kernel (4/5), new random ports
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/lullaby/.local/lib/python3.10/site-packages/PARIKernel/__main__.py", line 4, in <module>
    IPKernelApp.launch_instance(kernel_class=PARIKernel)
  File "/usr/lib/python3.10/site-packages/traitlets/config/application.py", line 845, in launch_instance
    app.initialize(argv)
  File "/usr/lib/python3.10/site-packages/traitlets/config/application.py", line 88, in inner
    return method(app, *args, **kwargs)
  File "/usr/lib/python3.10/site-packages/ipykernel/kernelapp.py", line 647, in initialize
    self.init_kernel()
  File "/usr/lib/python3.10/site-packages/ipykernel/kernelapp.py", line 499, in init_kernel
    kernel = kernel_factory(parent=self, session=self.session,
  File "/usr/lib/python3.10/site-packages/traitlets/config/configurable.py", line 540, in instance
    inst = cls(*args, **kwargs)
  File "PARIKernel/kernel.pyx", line 105, in PARIKernel.kernel.PARIKernel.__init__
  File "PARIKernel/io.pyx", line 67, in PARIKernel.io.PARIKernelIO.__init__
  File "/usr/lib/python3.10/site-packages/ipykernel/iostream.py", line 351, in __init__
    warnings.warn(
TypeError: argument for warn() given by name ('stacklevel') and position (3)
```
It seems like there is an error with launch_instance method of IPKernelApp. Maybe my error is specific to python 3.10 ?

**Comment by videlec** [comment]
Thanks for your report.

Note that the error does not come from pari-jupyter but `iostream.py`. This error could have been fixed in 926289f. Did you use the pari-jupyter from PyPI or the sources from github?

**Comment by dlesnoff** [comment]
I made a `pip install`. So I guess PyPI (I did not clone the github repo).
I will try with the repo this evening.

**Comment by videlec** [comment]
pip can clone it for you with
```
pip install git+https://github.com/sagemath/pari-jupyter
```

**Comment by dlesnoff** [comment]
I made a first pip install clone, a pip uninstall pari-jupyter (to remove the first package) then retried pip install (clone) but got the following error during the installation :
Building wheels for collected packages: pari-jupyter
  Building wheel for pari-jupyter (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-jhtwplb2/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-jhtwplb2/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-jfvtrpz9
       cwd: /tmp/pip-req-build-jhtwplb2/
  Complete output (16 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.10
  creating build/lib.linux-x86_64-3.10/PARIKernel
  copying PARIKernel/__main__.py -> build/lib.linux-x86_64-3.10/PARIKernel
  copying PARIKernel/__init__.py -> build/lib.linux-x86_64-3.10/PARIKernel
  running build_ext
  building 'PARIKernel.kernel' extension
  creating build/temp.linux-x86_64-3.10
  creating build/temp.linux-x86_64-3.10/PARIKernel
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fPIC -I/usr/include/python3.10 -c PARIKernel/kernel.c -o build/temp.linux-x86_64-3.10/PARIKernel/kernel.o
  cc1: fatal error: PARIKernel/kernel.c: No such file or directory
  compilation terminated.
  error: command '/usr/bin/gcc' failed with exit code 1
  ----------------------------------------

**Comment by videlec** [comment]
I believe you do not have cython installed (necessary to install from source). It is the tool that converts the `.pyx` files into `.c` file.

**Comment by dlesnoff** [comment]
Indeed, installed cython, pari-jupyter, and kernel works !

**Comment by videlec** [comment]
Thanks for your report and test. I will make a new release on PyPI to include that fix.

**Comment by dlesnoff** [comment]
You might want to update README.md to mark Cython as required dependency.

**Comment by aweisse** [comment]
I still have this issue when I install pari_jupyter (1.4.0) in Sage 9.7 via `sage -i pari_jupyter`.  It's a fresh install from source of Sage 9.7. Cython is present in this case as a standard package (0.29.32.p1). 

Here's a trace:
```
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/opt/sage/9.7/local/var/lib/sage/venv-python3.9/lib/python3.9/site-packages/PARIKernel/__main__.py", line 4, in <module>
    IPKernelApp.launch_instance(kernel_class=PARIKernel)
  File "/opt/sage/9.7/local/var/lib/sage/venv-python3.9/lib/python3.9/site-packages/traitlets/config/application.py", line 975, in launch_instance
    app.initialize(argv)
  File "/opt/sage/9.7/local/var/lib/sage/venv-python3.9/lib/python3.9/site-packages/traitlets/config/application.py", line 110, in inner
    return method(app, *args, **kwargs)
  File "/opt/sage/9.7/local/var/lib/sage/venv-python3.9/lib/python3.9/site-packages/ipykernel/kernelapp.py", line 647, in initialize
    self.init_kernel()
  File "/opt/sage/9.7/local/var/lib/sage/venv-python3.9/lib/python3.9/site-packages/ipykernel/kernelapp.py", line 499, in init_kernel
    kernel = kernel_factory(parent=self, session=self.session,
  File "/opt/sage/9.7/local/var/lib/sage/venv-python3.9/lib/python3.9/site-packages/traitlets/config/configurable.py", line 551, in instance
    inst = cls(*args, **kwargs)
  File "PARIKernel/kernel.pyx", line 105, in PARIKernel.kernel.PARIKernel.__init__
  File "PARIKernel/io.pyx", line 67, in PARIKernel.io.PARIKernelIO.__init__
  File "/opt/sage/9.7/local/var/lib/sage/venv-python3.9/lib/python3.9/site-packages/ipykernel/iostream.py", line 351, in __init__
    warnings.warn(
TypeError: argument for warn() given by name ('stacklevel') and position (3)
```

**Comment by aweisse** [comment]
Ok, I can fix the problem by upgrading to 1.4.1 with `sage -pip install --upgrade pari_jupyter`.
